dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/FaultToleranceCDIComponent.java

```diff
@@ -14,30 +14,32 @@
 
 import static java.util.Collections.unmodifiableList;
 import static org.osgi.service.component.annotations.ConfigurationPolicy.IGNORE;
+import static org.osgi.service.component.annotations.ReferenceCardinality.AT_LEAST_ONE;
 import static org.osgi.service.component.annotations.ReferenceCardinality.MANDATORY;
 import static org.osgi.service.component.annotations.ReferencePolicy.DYNAMIC;
 import static org.osgi.service.component.annotations.ReferencePolicyOption.GREEDY;
 
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
 
+import com.ibm.ws.microprofile.faulttolerance.spi.FTAnnotationInspector;
+import com.ibm.ws.microprofile.faulttolerance.spi.MetricRecorderProvider;
+
 import org.osgi.service.component.annotations.Activate;
 import org.osgi.service.component.annotations.Component;
 import org.osgi.service.component.annotations.Deactivate;
 import org.osgi.service.component.annotations.Reference;
 import org.osgi.service.component.annotations.ReferenceCardinality;
 
-import com.ibm.ws.microprofile.faulttolerance.spi.FTAnnotationInspector;
-import com.ibm.ws.microprofile.faulttolerance.spi.MetricRecorderProvider;
-
 /**
  * This component provides access to various services through a static methods
  */
 @Component(configurationPolicy = IGNORE, immediate = true)
 public class FaultToleranceCDIComponent {
 
-    private static MetricRecorderProvider metricProvider = null;
+    private static List<MetricRecorderProvider> metricProviders = new ArrayList<MetricRecorderProvider>();
```

Comment:
I _think_ this should probably be a `CopyOnWriteArrayList` - we don't expect regular writes but we want to avoid the possibility of changing the list while something else is iterating through it.

Reply:
Given how rare modifications will be I think this makes sense.